src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private Node[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key == null ? 0 : key.hashCode(), null, key, value);
+        if (table[getBucket(key)] == null) {
+            if (checkSizeOfTable()) {
```

Comment:
move resize check to the top of this method and do it only one time
this will make your code simpler

Reply:
you need to redo your `put()` with resize check on the top of the method